surefire-report-parser/src/main/java/org/apache/maven/plugins/surefire/report/TestSuiteXmlParser.java

```diff
@@ -217,6 +241,38 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
                     throw new SAXException("Failed to parse time value", e);
                 }
                 break;
+            case "flakyFailure":
+                testCase.addFlakyFailure(testCaseFlakyFailure);
+                testCaseFlakyFailure = null;
+                break;
+            case "flakyError":
+                testCase.addFlakyError(testCaseFlakyError);
+                testCaseFlakyError = null;
+                break;
+            case "stackTrace":
+                if (testCaseFlakyFailure != null) {
+                    testCaseFlakyFailure.setStackTrace(currentElement.toString());
+                }
+                if (testCaseFlakyError != null) {
+                    testCaseFlakyError.setStackTrace(currentElement.toString());
+                }
+                break;
+            case "system-out":
+                if (testCaseFlakyFailure != null) {
+                    testCaseFlakyFailure.setSystemOut(currentElement.toString());
+                }
+                if (testCaseFlakyError != null) {
+                    testCaseFlakyError.setSystemOut(currentElement.toString());
+                }
+                break;
+            case "system-err":
+                if (testCaseFlakyFailure != null) {
+                    testCaseFlakyFailure.setSystemErr(currentElement.toString());
+                }
+                if (testCaseFlakyError != null) {
+                    testCaseFlakyError.setSystemErr(currentElement.toString());
+                }
```

Comment:
This will be a problem when https://github.com/apache/maven-surefire/pull/692 is merged.

Reply:
Will it? There will be a conflict but I think it's just a matter of adding it to test case if it can't be added to the two others.

Does it make sense or do you foresee something more complex.

In any case, no problem to rebase this one or help finishing the other one. I really need this in so I'll do my best to help.

Thanks!

Reply:
I cannot exactly tell, but I see in the schema:
https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report.xsd multiple system out/err. We must make sure that the exact one is pulled in. Yet another issue is similar to https://issues.apache.org/jira/browse/SUREFIRE-2212. If those std streams are huge we should be able to disable this. See: https://github.com/apache/maven-surefire/pull/670. This should be opt-in. 

WDYT? I am willing to merge when all questions and test cases have been clarified and verified.

Reply:
Given the parsing is sequential, I don't think it's going to be hard to affect things properly.

But if we think parsing system out and system err is controversial and require more thinking, I can remove them from this patch. I added them to be comprehensive but atm I don't actually need them.

Reply:
Yes, please do so. It should be handled separate will adequate testing.

Reply:
I pushed an additional commit to drop the parsing of system out and system err.